File: sdk/SDK.Examples/src/FieldValidationExample.cs
using System;
using System.IO;
using Silanis.ESL.SDK;
using Silanis.ESL.SDK.Builder;

namespace SDK.Examples
{
    public class FieldValidationExample : SDKSample
    {
        public static void Main(string[] args)
        {
            new FieldValidationExample().Run();
        }

        public readonly string DOCUMENT_NAME = "My Document";

        public readonly string FIELD_NUMERIC_ID = "numeric";
        public readonly int FIELD_NUMERIC_MAX_LENGTH = 10;
        public readonly string FIELD_NUMERIC_ERROR_MESSAGE = "This field is not numeric";

        public readonly string FIELD_ALPHABETIC_ID = "alphabetic";
        public readonly int FIELD_ALPHABETIC_MIN_LENGTH = 3;
        public readonly int FIELD_ALPHABETIC_MAX_LENGTH = 10;
        public readonly string FIELD_ALPHABETIC_ERROR_MESSAGE = "This field is not alphabetic";

        public readonly string FIELD_ALPHANUMERIC_ID = "alphanumeric";
        public readonly int FIELD_ALPHANUMERIC_MIN_LENGTH = 5;
        public readonly string FIELD_ALPHANUMERIC_ERROR_MESSAGE = "This field is not alphanumeric";

        public readonly string FIELD_URL_ID = "url";
        public readonly String FIELD_URL_ERROR_MESSAGE = "The value in this field is not a valid URL";

        public readonly string FIELD_EMAIL_ID = "email";
        public readonly string FIELD_EMAIL_ERROR_MESSAGE = "The value in this field is not an email address";

        public readonly string FIELD_BASIC_ID = "basic";
        public readonly String FIELD_BASIC_OPTION_1 = "one";
        public readonly String FIELD_BASIC_OPTION_2 = "two";

        public readonly string FIELD_REGEX_ID = "regex";
        public readonly string FIELD_REGEX = "^[0-9a-zA-Z]+$";
        public readonly String FIELD_REGEX_ERROR_MESSAGE = "The value in this field does not match the expression";

        public string Email1
        {
            get
            {
                return email1;
            }
        }

        override public void Execute()
        {
            DocumentPackage package = PackageBuilder.NewPackageNamed(PackageName)
					.DescribedAs("This is a new package")
					.WithSigner(SignerBuilder.NewSignerWithEmail(email1)
					            .WithFirstName("John")
					            .WithLastName("Smith"))
                    .WithDocument(DocumentBuilder.NewDocumentNamed(DOCUMENT_NAME)
                                .FromStream(fileStream1, DocumentType.PDF)
				                .WithSignature(SignatureBuilder.SignatureFor(email1)
					              		.OnPage(0)
					               		.AtPosition(500, 100)
					               		.WithField(FieldBuilder.TextField()
                                            .WithId(FIELD_ALPHABETIC_ID)   				
                                            .OnPage(0)
					           				.AtPosition(500, 200)
					           				.WithValidation(FieldValidatorBuilder.Alphabetic()
                                                .MaxLength(FIELD_ALPHABETIC_MAX_LENGTH)
                                                .MinLength(FIELD_ALPHABETIC_MIN_LENGTH)
					                			.Required()
                                                .WithErrorMessage(FIELD_ALPHABETIC_ERROR_MESSAGE)))
					               		.WithField(FieldBuilder.TextField()
                                            .WithId(FIELD_NUMERIC_ID)
					           				.OnPage(0)
					           				.AtPosition(500, 300)
					           				.WithValidation(FieldValidatorBuilder.Numeric()					                			
                                                .WithErrorMessage(FIELD_NUMERIC_ERROR_MESSAGE)
                                                .Disabled()))
					               		.WithField(FieldBuilder.TextField()
                                            .WithId(FIELD_ALPHANUMERIC_ID)
					           				.OnPage(0)
					           				.AtPosition(500, 400)
					           				.WithValidation(FieldValidatorBuilder.Alphanumeric()					                			
                                                .WithErrorMessage(FIELD_ALPHANUMERIC_ERROR_MESSAGE)))
					               		.WithField(FieldBuilder.TextField()
                                            .WithId(FIELD_EMAIL_ID)
							           		.OnPage(0)
							           		.AtPosition(500, 500)
							           		.WithValidation(FieldValidatorBuilder.Email()					                			
                                                .WithErrorMessage(FIELD_EMAIL_ERROR_MESSAGE)))
					               		.WithField(FieldBuilder.TextField()
                                            .WithId(FIELD_URL_ID)
					           				.OnPage(0)
					           				.AtPosition(500, 600)
					           				.WithValidation(FieldValidatorBuilder.URL()
                                                .WithErrorMessage(FIELD_URL_ERROR_MESSAGE)))
					               		.WithField(FieldBuilder.TextField()
                                            .WithId(FIELD_REGEX_ID)
					           				.OnPage(0)
					           				.AtPosition(500, 700)
                                            .WithValidation(FieldValidatorBuilder.Regex(FIELD_REGEX)
                                                .WithErrorMessage(FIELD_REGEX_ERROR_MESSAGE)))))
					.Build();

            packageId = eslClient.CreatePackage(package);

            eslClient.SendPackage(packageId);
            retrievedPackage = eslClient.GetPackage(packageId);
        }
    }
}

File: sdk/SDK.VBExamples/SDK.VBExamples/FieldValidationExample.vb
Imports System
Imports System.IO
Imports Silanis.ESL.SDK
Imports Silanis.ESL.SDK.Builder

Namespace SDK.VBExamples
    Public Module FieldValidationExample
        Dim apiToken = "YOUR TOKEN HERE"
        Dim baseUrl = "ENVIRONMENT URL HERE"

        Sub Main()
            ' Create new esl client with api token and base url
            Dim client = New EslClient(apiToken, baseUrl)
            Dim file = New FileInfo(Directory.GetCurrentDirectory() + My.Resources.document)

            Dim package = PackageBuilder.NewPackageNamed("Fields example " + DateTime.Now) _
                .DescribedAs("This is a new package") _
                .WithSigner(SignerBuilder.NewSignerWithEmail("john.smith@email.com") _
                    .WithFirstName("John") _
                    .WithLastName("Smith")) _
                .WithDocument(DocumentBuilder.NewDocumentNamed("My Document") _
                    .FromFile(file.FullName) _
                    .WithSignature(SignatureBuilder.SignatureFor("john.smith@email.com") _
                        .OnPage(0) _
                        .AtPosition(500, 100) _
                        .WithField(FieldBuilder.TextField() _
                            .OnPage(0) _
                            .AtPosition(500, 200) _
                            .WithValidation(FieldValidatorBuilder.Alphabetic() _
                                .MaxLength(10) _
                                .MinLength(3) _
                                .Required() _
                                .WithErrorMessage("This field is not alphabetic"))) _
                        .WithField(FieldBuilder.TextField() _
                            .OnPage(0) _
                            .AtPosition(500, 300) _
                            .WithValidation(FieldValidatorBuilder.Numeric() _
                                .WithErrorMessage("This field is not numeric"))) _
                        .WithField(FieldBuilder.TextField() _
                            .OnPage(0) _
                            .AtPosition(500, 400) _
                            .WithValidation(FieldValidatorBuilder.Alphanumeric() _
                                .WithErrorMessage("This field is not alphanumeric"))) _
                        .WithField(FieldBuilder.TextField() _
                            .OnPage(0) _
                            .AtPosition(500, 500) _
                            .WithValidation(FieldValidatorBuilder.Email() _
                                .WithErrorMessage("The value in this field is not an email address"))) _
                        .WithField(FieldBuilder.TextField() _
                            .OnPage(0) _
                            .AtPosition(500, 600) _
                            .WithValidation(FieldValidatorBuilder.URL() _
                                .WithErrorMessage("The value in this field is not a valid URL"))) _
                        .WithField(FieldBuilder.TextField() _
                            .OnPage(0) _
                            .AtPosition(500, 700) _
                            .WithValidation(FieldValidatorBuilder.Regex("^[0-9a-zA-Z]+$") _
                                .WithErrorMessage("The value in this field does not match the expression"))))) _
                .Build()

            Dim id = client.CreatePackage(package)

            client.SendPackage(id)

            Console.WriteLine("Package {0} was sent", id.Id)
        End Sub
    End Module
End Namespace